### PR TITLE
change listReleases to accept query params instead of body

### PR DESCRIPTION
**Pull request by christiangebhard**
Adjust release.go to accept query params instead of json body (+ error handling).
Tested locally with some requests.

### CODE REVIEW

File: releases.go [hunk 1]
```diff
@@ -562,11 +562,85 @@ func listReleases(c *gin.Context) {
...
+		return
+	}
+
+	options.Limit, err = strconv.Atoi(c.DefaultQuery("limit", "0"))
+	if err != nil {
+		respErr(c, err)
+		return
+	}
+
+	options.Offset, err = strconv.Atoi(c.DefaultQuery("offset", "0"))
```
Comment (miriSch) [inline releases.go:596]: For the limit and the offset there are no default values in the documentation. Do they have to be set? Does a default of 0 make sense for the limit?

File: releases.go [hunk 2]
```diff
@@ -562,11 +562,85 @@ func listReleases(c *gin.Context) {
...
+	if err != nil {
+		respErr(c, err)
+		return
+	}
+
+	options.Filter = c.Query("filter")
```
Comment (miriSch) [inline releases.go:602]: According to the documentation, default is an empty string. Does that matter?

File: releases.go (hunk 1 above)
Comment (christiangebhard) [reply releases.go:596]: The original function where it has been taken from the json body defaulted it to 0.

File: releases.go (hunk 2 above)
Comment (christiangebhard) [reply releases.go:602]: If the filter param is not set `c.Query` will return ""

File: releases.go (hunk 1 above)
Comment (christiangebhard) [reply releases.go:596]: And if I understand the client code correctly, the limit only gets applied if > 0.

**Review by miriSch — APPROVED** [review]

**Pull request merged**